recipes/css-selector.yaml

```diff
@@ -0,0 +1,15 @@
+data:
+  - title
+  - short_title?
+  - mdn_url
+body:
+  - prose.short_description
+  - prose.syntax
+  - prose.description?
+  - prose.applicable_properties?
```

Comment:
I think we should go ahead and make this `data.applicable_properties`. It'll be very similar to `data.constituent_properties` so we may as well do it

Reply:
https://wiki.developer.mozilla.org/en-US/docs/Web/CSS/::first-letter#Allowable_properties, for example, presents properties in groups - font related, background-related, and so on.

If we specify it like `data.constituent_properties` then it's a list of links, 50-odd items, and no grouping. Is this OK or is it a significant loss? Also that page wants to annotate some links:

> `vertical-align` (only if float is none)

...so we'd either have to support that or lose this information.

Reply:
The more correct thing here would be to do some sort of user research to find out how developers use this information. Are they using it as a troubleshooting tool after the fact ("Why doesn't this property work? Ah, OK, it's in this list…") or are they reading it first? That'd probably tell us if groups or a list is more useful.

But going only on instinct here, I'd say that I don't much care for the grouped style. As a writer, I think I'd prefer some semi-free-form intro text that says something like, "Only a subset of CSS properties work with this pseudo-element. Supported properties include font, background, margin, padding, border, and text decoration properties" followed by an alphabetically sorted list of properties. That feels way more maintainable than trying to figure out which group (or which miscellaneous group, as in the `::first-letter` example) a property goes into.

As a reader, I think I'd prefer the list too. I don't trust myself to know what properties are in what groups, the property names are more widely recognizable than group descriptions, and I can skim a list faster than the current format.

For the annotations, I'm OK with losing them:

* The cue related pseudo-classes have some interesting behavior to do with borders and forming boxes. That seems notable enough for its own `prose.*` section.
* `::first-letter`'s float annotation is a reminder of a general rule about `vertical-align`: it only applies to inline-level elements. If `float` is not none, then it's not inline. The other one is "`word-spacing` (when appropriate)", which seems pretty pointless.
* `::selection` has a call out box for not supporting `background-image`. Assuming that was put in because it surprises people, that's a little loss, but I don't think we need to handle annotations generally for this one entry.

Reply:
Thanks for this thoughtful response!

> The more correct thing here would be to do some sort of user research to find out how developers use this information. Are they using it as a troubleshooting tool after the fact ("Why doesn't this property work? Ah, OK, it's in this list…") or are they reading it first? That'd probably tell us if groups or a list is more useful.

In the absence of this my uninformed speculation is option 1, just because noone reads documentation until something doesn't work. But I definitely take your point that a list is easier to scan and doesn't make the assumption that our categorization of things matches the reader's.